k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java

```diff
@@ -132,6 +104,38 @@ void populateFromGetMessageCursor(Cursor cursor) throws MessagingException {
         headerNeedsUpdating = false;
     }
 
+    void populateUidAndFlags(Cursor cursor) throws MessagingException {
+        setUid(cursor.getString(cursor.getColumnIndex("uid")));
+        String flagList = cursor.getString(cursor.getColumnIndex("flags"));
+        if (flagList != null && flagList.length() > 0) {
+            String[] flags = flagList.split(",");
+
+            for (String flag : flags) {
+                try {
+                    this.setFlagInternal(Flag.valueOf(flag), true);
+                }
+
+                catch (Exception e) {
+                    if (!"X_BAD_FLAG".equals(flag)) {
+                        Timber.w("Unable to parse flag %s", flag);
+                    }
+                }
+            }
+        }
+
+        boolean deleted = (cursor.getInt(cursor.getColumnIndex("deleted")) == 1);
+        boolean read = (cursor.getInt(cursor.getColumnIndex("read")) == 1);
+        boolean flagged = (cursor.getInt(cursor.getColumnIndex("flagged")) == 1);
+        boolean answered = (cursor.getInt(cursor.getColumnIndex("answered")) == 1);
+        boolean forwarded = (cursor.getInt(cursor.getColumnIndex("forwarded")) == 1);
+
+        setFlagInternal(Flag.DELETED, deleted);
```

Comment:
Anyone know if it's possible to (for instance) have deleted in the database as 0 but DELETED in the `flags` column`?